tests/jax/components/building/parameter_client_test.py

```diff
@@ -0,0 +1,315 @@
+    """
+
+    mock_builder = mock_builder_with_parameter_client
+    mock_builder.store.is_evaluator = False
+    exec_param_client = ExecutorParameterClient(
+        config=ExecutorParameterClientConfig(executor_parameter_update_period=100)
+    )
+    exec_param_client.on_building_executor_parameter_client(builder=mock_builder)
+
+    assert mock_builder.store.executor_parameter_client._all_keys == [
+        "evaluator_episodes",
+        "evaluator_steps",
+        "executor_episodes",
+        "executor_steps",
+        "networks-network_agent_0",
+        "networks-network_agent_1",
+        "networks-network_agent_2",
+        "trainer_steps",
+        "trainer_walltime",
+    ]
+    assert mock_builder.store.executor_parameter_client._get_keys == [
+        "networks-network_agent_0",
+        "networks-network_agent_1",
+        "networks-network_agent_2",
+        "trainer_steps",
+        "trainer_walltime",
+        "evaluator_steps",
+        "evaluator_episodes",
+        "executor_episodes",
+        "executor_steps",
+    ]
+    assert mock_builder.store.executor_parameter_client._set_keys == [
+        "executor_episodes",
+        "executor_steps",
+    ]
+    assert mock_builder.store.executor_parameter_client._parameters == {
+        "networks-network_agent_0": {"weights": 0, "biases": 0},
+        "networks-network_agent_1": {"weights": 1, "biases": 1},
+        "networks-network_agent_2": {"weights": 2, "biases": 2},
+        "trainer_steps": np.array(0, dtype=np.int32),
+        "trainer_walltime": np.array(0.0, dtype=np.float32),
+        "evaluator_steps": np.array(0, dtype=np.int32),
+        "evaluator_episodes": np.array(0, dtype=np.int32),
+        "executor_episodes": np.array(0, dtype=np.int32),
+        "executor_steps": np.array(0, dtype=np.int32),
+    }
+    assert mock_builder.store.executor_parameter_client._get_call_counter == 0
+    assert mock_builder.store.executor_parameter_client._set_call_counter == 0
+    assert mock_builder.store.executor_parameter_client._set_get_call_counter == 0
+    assert mock_builder.store.executor_parameter_client._update_period == 100
+    assert isinstance(
+        mock_builder.store.executor_parameter_client._client, ParameterServer
+    )
+
+
+def test_executor_parameter_client_evaluator_with_parameter_client(
+    mock_builder_with_parameter_client: Builder,
+) -> None:
+    """Test evaluator parameter client.
+
+    Args:
+        mock_builder_with_parameter_client: mava builder object
+    """
+
+    mock_builder = mock_builder_with_parameter_client
+    mock_builder.store.is_evaluator = True
+    exec_param_client = ExecutorParameterClient(
+        config=ExecutorParameterClientConfig(executor_parameter_update_period=100)
+    )
+    exec_param_client.on_building_executor_parameter_client(builder=mock_builder)
+
+    assert hasattr(mock_builder.store, "executor_parameter_client")
+
+    assert mock_builder.store.executor_parameter_client._all_keys == [
```

Comment:
Everything below seems the same as above test. Do you need to retest it? I would imagine you're testing the same code, could you just remove it? If I'm wrong and it's different code generating the same data then feel free to ignore this

Reply:
It is indeed different (although slightly) since the below case is for an executor to be used for evaluation and the above was for a 'vanilla' executor.